### ISSUE TITLE
FMA implementation subtly incorrect

### ISSUE THREAD

**Issue by gnzlbg** [rust-lang/libm#200]
It appears that unsigned integer subtraction was mapped to `Sub::sub(a, b)` in Rust (`a - b`), instead of `a.wrapping_sub(b)`. In C, unsigned integer subtraction wraps, but in rust, unless `wrapping_sub` is used, it panics on overflow in debug builds, and has UB in release builds.

**Comment by Lokathor** [comment]
It is not UB in release builds, it is wrapping in release builds.

**Comment by gnzlbg** [comment]
Indeed. So something else must be off as well, because it was producing segfaults for me in release.

**Comment by tgross35** [comment]
I believe https://github.com/rust-lang/libm/pull/267 should have fixed this. In any case, I haven't seen it reproduced against the MPFR tests.
